cst438_gradebook/src/test/java/com/cst438/AssignmentControllerTests.java

```diff
@@ -0,0 +1,184 @@
+        Assignment assign1 = new Assignment();
+        assign1.setId(1);
+        assign1.setName("Rest assignment");
+        assign1.setDueDate(Date.valueOf("2023-09-23"));
+
+        Course course1 = new Course();
+        course1.setTitle("Software Engineering ");
+        course1.setCourse_id(1);
+        assign1.setCourse(course1);
+
+        Assignment assign2 = new Assignment();
+        assign2.setId(2);
+        assign2.setName("Modus tollens");
+        assign2.setDueDate(Date.valueOf("2023-10-27"));
+
+        Course course2 = new Course();
+        course2.setTitle("Reasoning with Logic");
+        course2.setCourse_id(2);
+        assign2.setCourse(course2);
+        List<Assignment> assignments = Arrays.asList(assign1, assign2);
+
+        when(assignmentRepository.findByEmail(instructorEmail)).thenReturn(assignments);
+        MockHttpServletResponse response = mvc.perform(MockMvcRequestBuilders.get("/assignment")
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn().getResponse();
+
+        AssignmentDTO[] res = new AssignmentDTO[assignments.size()];
+        for (int i = 0; i < assignments.size(); i++) {
+            Assignment assignment = assignments.get(i);
+            res[i] = new AssignmentDTO(
+                    assignment.getId(),
+                    assignment.getName(),
+                    assignment.getDueDate().toString(),
+                    assignment.getCourse().getTitle(),
+                    assignment.getCourse().getCourse_id()
+            );
+        }
+        assertEquals(2, res.length);
+        assertEquals("Rest assignment", res[0].assignmentName());
+        assertEquals("Modus tollens", res[1].assignmentName());
+        assertEquals(200, response.getStatus());
+    }
+    @Test
+    public void UpdateAssignmentTest() throws Exception {
+
+        AssignmentDTO assignmentDTO = new AssignmentDTO(1, "Updated Assignment", "2023-09-23", "CST 363 - Introduction to Database Systems", 31045);
+        Assignment existingAssignment = new Assignment();
+        existingAssignment.setId(1);
+        existingAssignment.setName("Original Assignment");
+        existingAssignment.setDueDate(Date.valueOf("2023-09-28"));
+
+        when(assignmentRepository.findById(1)).thenReturn(Optional.of(existingAssignment));
+        when(courseRepository.findById(3)).thenReturn(Optional.of(new Course()));
+        mvc.perform(put("/assignment/{assignment_id}", 1)
+                        .content(asJsonString(assignmentDTO))
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(content().string(""));
+    }
+
+    private static String asJsonString(final Object obj) {
+        try {
+
+            return new ObjectMapper().writeValueAsString(obj);
```

Comment:
These are the functions which I was referring to, which should be used to transform json to object and vice versa for assert statement in the code after returning from the controller mapping.